webhooks/services/billing_processor.py

```diff
@@ -0,0 +1,275 @@
+        """
+        Post billing event to LAGO
+        """
+        logger.debug(f"Posting billing event: {event}")
+        try:
+            lago_response = await self._client.post(
+                url=self._LAGO_URL,
+                json={"event": event.model_dump()},
+            )
+
+            logger.error(f"Response from LAGO: {lago_response.json()}")
+            lago_response.raise_for_status()
+
+        except HTTPException as e:
+            if e.status_code == 422:
+                logger.error(f"Error posting billing event >>> : {e.detail}")
+                pass
+            logger.error(f"Error posting billing event: {e}")
+            raise e
+
+    def _convert_credential_event(
+        self, event: Dict[str, Any]
+    ) -> CredentialBillingEvent:
+        """
+        Convert credential event to LAGO event
+        """
+        logger.debug(f"Converting credential event: {event}")
+
+        # using thread_id as transaction_id
+        payload: Dict[str, Any] = event.get("payload")
+        lago_event = CredentialBillingEvent(
+            transaction_id=payload.get("thread_id"),
+            external_customer_id=event.get("group_id"),
+        )
+        return lago_event
+
+    def _convert_proofs_event(self, event: Dict[str, Any]) -> ProofBillingEvent:
+        """
+        Convert proofs event to LAGO event
+        """
+        logger.debug(f"Converting proofs event: {event}")
+
+        # using thread_id as transaction_id
+        payload: Dict[str, Any] = event.get("payload")
+        lago_event = ProofBillingEvent(
+            transaction_id=payload.get("thread_id"),
+            external_customer_id=event.get("group_id"),
+        )
+        return lago_event
+
+    def _convert_endorsements_event(
+        self, event: Dict[str, Any]
+    ) -> EndorsementBillingEvent:
+        """
+        Convert endorsements event to LAGO event
+        """
+        logger.debug(f"Converting endorsements event: {event}")
+
+        payload: Dict[str, Any] = event.get("payload")
+
+        # use operation type to determine the endorsement type
+        # using transaction_id asfor LAGO transaction_id
+        if payload.get("type") == "100":
+            lago_event = AttribBillingEvent(
+                transaction_id=payload.get("transaction_id"),
+                external_customer_id=event.get("group_id"),
+            )
+            return lago_event
+
+        if payload.get("type") == "102":
+            lago_event = CredDefBillingEvent(
+                transaction_id=payload.get("transaction_id"),
+                external_customer_id=event.get("group_id"),
+            )
+            return lago_event
+
+        if payload.get("type") == "113":
+            lago_event = RevRegDefBillingEvent(
+                transaction_id=payload.get("transaction_id"),
+                external_customer_id=event.get("group_id"),
+            )
+            return lago_event
```

Comment:
Veeeery similar method operations happening here. The only difference is the class being constructed. We are defining `payload.get("transaction_id")` each time. Rather just do that once at the beginning. Or maybe GPT has a more elegant way to define classes dynamically which have the same attributes.

Actually, it makes me wonder why different classes is necessary ... just because they have the Literal attribute? Maybe we should just define that dynamically here, and then there's one Class being constructed, but with one of many codes. Idk, might be more readable that way.